Sources/Jetworking/Client/Client.swift

```diff
@@ -375,27 +385,36 @@ public final class Client {
         }
     }
 
+    private func handleSuccessVoid<ResponseType>(currentURLResponse: HTTPURLResponse, endpoint: Endpoint<ResponseType>?, data: Data?, completion: @escaping RequestCompletion<ResponseType>) {
+        if ResponseType.self is Void.Type {
+            return enqueue(completion(currentURLResponse, .success(() as! ResponseType)))
+        }
+
+        enqueue(completion(currentURLResponse, .failure(APIError.unexpectedError)))
+    }
+
+    func handleSuccessDecodable<ResponseType: Decodable>(currentURLResponse: HTTPURLResponse, endpoint: Endpoint<ResponseType>?, data: Data?, completion: @escaping RequestCompletion<ResponseType>) {
+        let decoder: Decoder = endpoint?.decoder ?? configuration.decoder
+
+        guard let decodedData = try? decodeResponse(ResponseType.self, with: decoder, from: data) else {
+            return enqueue(completion(currentURLResponse, .failure(APIError.decodingError)))
+        }
+        // Evaluate Header fields --> urlResponse.allHeaderFields
+
+        enqueue(completion(currentURLResponse, .success(decodedData)))
+    }
+
     private func enqueue(_ completion: @escaping @autoclosure () -> Void) {
         configuration.responseQueue.async {
             completion()
         }
     }
 
     /// Decodes an instance of the given type.
-    private func decodeResponse<ResponseType: Decodable>(_ dataType: ResponseType.Type, from data: Data?) throws -> ResponseType {
+    private func decodeResponse<ResponseType>(_ dataType: ResponseType.Type, with decoder: Decoder, from data: Data?) throws -> ResponseType where ResponseType: Decodable {
```

Comment:
Do we still need this function since it is just 2 lines and only used in handleSuccessDecodable?

Reply:
This method is now removed.